apps/argus/src/services/controller_service.rs

```diff
@@ -125,3 +161,595 @@ impl Service for ControllerService {
         Ok(())
     }
 }
+
+/// Determines if an on-chain price update is needed based on the latest Pyth price,
+/// the current on-chain price (if available), and the subscription's update criteria.
+#[tracing::instrument()]
+fn needs_update(
+    pyth_price: &Price,
+    chain_price_opt: Option<&Price>,
+    update_criteria: &UpdateCriteria,
+) -> bool {
+    // If there's no price currently on the chain for this feed, an update is always needed.
+    let chain_price = match chain_price_opt {
+        None => {
+            tracing::debug!("Update criteria met: No chain price available.");
```

Comment:
in price pusher sometimes i found these logs helpful, might not be bad to have them at info level, or maybe expose as metrics or trace